### DISCUSSION TITLE
How to show boot/update progress status?

### DISCUSSION THREAD

**Discussion by tdjastrzebski** [mcu-tools/mcuboot#1926] [Q&A]
I am looking for a way to implement a nice boot/update progress bar or at least text status, like those typically available on laptops during bios update or boot.
I found no way of attaching a callback external to MCUboot where continuous boot/update progress could be updated - just like a log streamed to uart. I could probably use another uart to read log streamed to the first one by MCUboot, but such solutions seems awkward - although it may work. There must be a better way.
Probably the only 'hook action' available in MCUboot is [mcuboot_status_change()](https://github.com/mcu-tools/mcuboot/blob/8dab2b975e14e00564eb62383448f70efd01f324/boot/bootutil/include/bootutil/mcuboot_status.h) but it does not provide progress info (e.g. phase 1 of 10, block 58 of 934 written) - just the general status enum.
Please advise

**Comment by JPHutchins** [comment]
This would of course be possible by patching a callback after each chunk is uploaded.  The callback could be called with the present offset as argument allowing for usage as a progress bar to a series of LEDs, an SMP event stream over UART, whatever.

I think that no one wants to officially support this "footgun callback" for obvious reasons!

We have a device that "blacks out" for like 90 seconds while it swaps from external SPI flash, decrypts, and verifies.  So we don't love the lack of visibility, but we do love that it works without bugs!
